setuptools/tests/test_wheel.py

```diff
@@ -115,11 +115,11 @@ def build_wheel(extra_file_defs=None, **kwargs):
 
 
 def tree_set(root):
-    contents = set()
-    for dirpath, dirnames, filenames in os.walk(root):
-        for filename in filenames:
-            contents.add(os.path.join(os.path.relpath(dirpath, root), filename))
-    return contents
+    return set(
+        os.path.join(os.path.relpath(dirpath, root), filename)
+        for dirpath, dirnames, filenames in os.walk(root)
+        for filename in filenames
+    )
```

Comment:
From personal experience, list comprehensions are just faster than generators if you know you'll iterate over all items.

```suggestion
    return set([
        os.path.join(os.path.relpath(dirpath, root), filename)
        for dirpath, dirnames, filenames in os.walk(root)
        for filename in filenames
    ])
```

Reply:
I have applied **C4** rules as well in this PR.

Reply:
Aren't you concerned enabling all of `flake-8 comprehensions` is gonna further delay the review of this PR?
I'm pretty sure @jaraco would prefer [`lint.flake8-comprehensions.allow-dict-calls-with-keyword-arguments = true`](https://docs.astral.sh/ruff/settings/#lint_flake8-comprehensions_allow-dict-calls-with-keyword-arguments) for [unnecessary-collection-call (C408)](https://docs.astral.sh/ruff/rules/unnecessary-collection-call/#unnecessary-collection-call-c408)

Reply:
Aren't you concerned enabling all of `flake-8 comprehensions` is gonna further delay the review of this PR?
For isntance, I'm pretty sure @jaraco would prefer [`lint.flake8-comprehensions.allow-dict-calls-with-keyword-arguments = true`](https://docs.astral.sh/ruff/settings/#lint_flake8-comprehensions_allow-dict-calls-with-keyword-arguments) for [unnecessary-collection-call (C408)](https://docs.astral.sh/ruff/rules/unnecessary-collection-call/#unnecessary-collection-call-c408)

Reply:
I'm not in a hurry, but I probably went overboard :smile: Should I apply `C401` and `C403` only from the C4 set?

Reply:
Imo it's probably safer to just recognize that this line can be improved further. Either with a single manual change, or as a follow-up to keep all changes here automated.

Especially if we provide a PR to enable `C4` separatly, this will kinda just fix itself.

But that's just my opinion from previous contributions, I'm not a maintainer ^^"

Reply:
Agreed. I'll remove `C4` rules and keep suboptimal automatic `FURB` fixes — they will be fixed by an upcoming PR enforcing `C4` rules.